backend/libs/parking/src/parking.service.ts

```diff
@@ -1,63 +1,332 @@
-import { Injectable } from '@nestjs/common';
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
 import { CommandBus, QueryBus } from "@nestjs/cqrs";
 import { AssignParkingCommand } from './commands/impl/assignParking.command';
 import { FreeParkingCommand } from './commands/impl/freeParking.command';
 import { ReserveParkingCommand } from './commands/impl/reserveParking.command';
 import { ParkingNotFound } from "./errors/parkingNotFound.error";
-import { getAvailableParkingQuery } from './queries/impl/getAvailableParking.query';
+import { GetAvailableParkingQuery } from './queries/impl/getAvailableParking.query';
+import { GetFreeParkingQuery } from './queries/impl/getFreeParking.query';
+import { UnreserveParkingCommand } from './commands/impl/unreserveParking.command';
+import { AddParkingCommand } from './commands/impl/addParking.command';
+import { ExternalError } from './errors/externalError.error';
+import { VisitorInviteService } from '@vms/visitor-invite/visitor-invite.service';
+import { InvalidQuery } from './errors/invalidQuery.error';
+import { GetInviteReservationQuery } from './queries/impl/getInviteReservation.query';
+import { CreateNParkingSpotsCommand } from './commands/impl/createNParkingSpots.command';
+import { InviteNotFound } from '@vms/visitor-invite/errors/inviteNotFound.error';
+import { ReservationNotFound } from './errors/reservationNotFound.error';
+import { GetReservationsQuery } from './queries/impl/getReservations.query';
+import { GetParkingReservationsQuery } from './queries/impl/getParkingReservations.query';
 
 @Injectable()
 export class ParkingService {
-    constructor(private commandBus: CommandBus, private queryBus: QueryBus) {}
+    constructor(private commandBus: CommandBus, 
+                private queryBus: QueryBus,
+                @Inject(forwardRef(() => VisitorInviteService))
+                private inviteService: VisitorInviteService) {}
 
-    async getAvailableParking(){
-        return this.queryBus.execute(
-            new getAvailableParkingQuery()
+    /*
+    Create more visitor parking
+
+    Throws: External error
+    Returns: new parking object
+    */
+    async addParking(
+    ){
+        const parking = await this.commandBus.execute(
+            new AddParkingCommand()
+        );
+
+        if(parking)
+            return parking;
+        else
+            //for testing purposes
+            throw new ExternalError("Error outside the parking.service");
```

Comment:
Please give these meaningful messages, describe the error being thrown. If it's for testing please remove before making a PR

Reply:
why

Reply:
The error message is not meaningful, what does the error message mean?

Reply:
We are still developing, Would be foolish to remove info that might help us debug in the future.

Reply:
Yes I agree, but the error is general and does not give me any information about what went wrong

Reply:
> Yes I agree, but the error is general and does not give me any information about what went wrong

At that stage that is all I know. I know not to search for errors within the service.ts